### DISCUSSION TITLE
want help with code, to draw a fullscreen quad

### DISCUSSION THREAD

**Discussion by kyoobey** [gfx-rs/wgpu#2568] [Q&A] [answered]
here is my full code: https://gist.github.com/tshrpl/01999d1244a92acb4812abede2a3ff3d

the result i get:
![Capture](https://user-images.githubusercontent.com/61683653/161118099-a9916949-41c1-4352-ab72-2b56bda052db.PNG)

expected result: a fullscreen quad

**Comment by kvark** [comment]
Your example is fairly large. I would recommend just taking hello-triangle, and growing it into what you need.

**Comment by kyoobey** [comment, marked as answer]
found the problem
its line 509 in main.rs https://gist.github.com/tshrpl/01999d1244a92acb4812abede2a3ff3d#file-main-rs-L509
centroid should be the average of _vertex_0 and _vertex_2 or _vertex_1 and _vertex_3 not _vertex_0 and _vertex_3
